tool/src/dotnet/Runtime.cs

```diff
@@ -0,0 +1,138 @@
+// Automatically generated by Diplomat
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace {{NAMESPACE}}.Diplomat;
+
+#nullable enable
+
+[UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+delegate void WriteableFlush(IntPtr self);
+
+[UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+[return: MarshalAs(UnmanagedType.U1)]
+delegate bool WriteableGrow(IntPtr self, nuint capacity);
+
+[Serializable]
+[StructLayout(LayoutKind.Sequential)]
+public struct DiplomatWriteable : IDisposable
```

Comment:
Yeah, totally agreed… but there is no native string type backed by UTF-8 in dotnet frameworks AFAIK. We could provide our own `Diplomat.Utf8String` type, but this is not trivial and mostly useless on its own because all other API are expecting the standard UTF-16 encoded `string`/`String` type. I expect API to become overall less ergonomic by forcing users to explicitly convert between UTF-16 and UTF-8 anyway. There isn’t even a way to re-use the unmanaged buffer as-is to get a managed `byte[]` as-is. At least, it wasn’t clear. Maybe we can open an issue to investigate further.

Reply:
Alright, this makes sense. Can you leave a comment to this effect explaining why this is the best option?

Reply:
Will do! :+1: